Aurora config

oksanaguba opened the pull request:
this branch will be eventually reworked into a branch with clean history

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Failure: Timed out waiting for job SCREAM_PullRequest_Autotester_Mappy to start: Total Wait = 1803
- Other jobs have been previously started - We must stop them...

ambrad committed:
Homme(xx)/SL: Enhanced trajectory method.

oksanaguba commented:
ERS test passing 
```
~/ess/ERS_Ln9.ne4pg2_ne4pg2.F2010-SCREAMv1.sunspot-pvc_oneapi-ifxgpu.scream-sunspot_run.C.20240715_234719_xed0zr 
```
```
onguba@uan-0001:~/runhomme/for-cprnc/utils/cime/CIME/non_py/cprnc> pwd
/home/onguba/runhomme/for-cprnc/utils/cime/CIME/non_py/cprnc
onguba@uan-0001:~/runhomme/for-cprnc/utils/cime/CIME/non_py/cprnc> ./cprnc /home/onguba/ess/ERS_Ln9.ne4pg2_ne4pg2.F2010-SCREAMv1.sunspot-pvc_oneapi-ifxgpu.scream-sunspot_run.C.20240715_234719_xed0zr/run/ERS_Ln9.ne4pg2_ne4pg2.F2010-SCREAMv1.sunspot-pvc_oneapi-ifxgpu.scream-sunspot_run.C.20240715_234719_xed0zr.cpl.hi.0001-01-01-32400.nc.base /home/onguba/ess/ERS_Ln9.ne4pg2_ne4pg2.F2010-SCREAMv1.sunspot-pvc_oneapi-ifxgpu.scream-sunspot_run.C.20240715_234719_xed0zr/run/ERS_Ln9.ne4pg2_ne4pg2.F2010-SCREAMv1.sunspot-pvc_oneapi-ifxgpu.scream-sunspot_run.C.20240715_234719_xed0zr.cpl.hi.0001-01-01-32400.nc.rest
```

xylar committed:
First draft of scaled DISMF module

xylar committed:
Hook up activation of scaledDISMFPKG package

xylar committed:
Add subroutine calls to ocean mct driver

xylar committed:
Add ocn_scaled_dismf module to cmake build

xylar committed:
Update history and running mean of removed ice runoff

xylar committed:
Update DISMF files to include total fluxes

Read in areaIntegAnnMeanDataIcebergIceShelfFreshwaterFlux from
these files

xylar committed:
Update DIB files to include total fluxes

Read in areaIntegAnnMeanDataIcebergIceShelfFreshwaterFlux from
these files

xylar committed:
Add scaling of DISMF by removed runoff

xylar committed:
Add support for scaling DIB

Most of the support is in place, just not the coupling between
MPAS-Ocean and MPAS-Seaice to pass `runningMeanRemovedIceRunoff`

xylar committed:
Add ocn-->ice coupling of runningMeanRemovedIceRunoff

xylar committed:
Remove trailing whitespace

xylar committed:
Add a stelth exact restart test

xylar committed:
Accumulate daily mean before daily alarm

Since the alarm rings at the end of a daily interval, we need
to accumulate the removed runoff before, not after, we reset
the daily accumulation.

xylar committed:
Move scaled DISMF update inside time stepping loop

This is necessary becasue we want to update before streams get
written.

xylar committed:
First zero entry in removed runoff history is zero

We want this to be a valid entry that contributes to the running
mean.

xylar committed:
Fix DISMF defaults for OCN_CO2_TYPE to match WCYCL

xylar committed:
Add runningMeanRemovedIceRunoff to global stats

mjschmidt271 committed:
improve handling of initial condition and add formatting to DAG indicating as such

fixed minor bug--still an issue with 2 fields in p3 for pg2 cases

print formatting fix for dag

remove 2 intermediate DAGs

add descriptor box for IC fields

cbegeman committed:
Add del2,del4 source terms to manufactured soln

rljacob reviewed (dismissed):
You should use/modify the aurora/sunspot configs added in https://github.com/E3SM-Project/E3SM/pull/6553/.  This creates too many machine definitions and doesn't follow the conventions we've established for machines with cpu/gpu nodes.

rljacob commented:
This PR can not go in as is.  And adding "scream" is also not a solution.

oksanaguba commented:
i cannot yet clean this branch till homme sycl changes are in scream repo.

xylar committed:
Update DIB and DISMF files for SOwISC12to30E3r3

xylar committed:
Shorten scaled_dib_dismf test to 5 days and use small layout

xylar committed:
Add PR number to test README

tcclevenger committed:
Create IOPForcing ATM process

Moves IOP forcing from within HOMME interface to it's own physics
process, called directly after dynamics process. Also, changes name of
IntensiveObservationPeriod class to IOPDataManager.

jonbob committed:
Make bld files consistent with Registry

xylar committed:
Reduce config_ais_ice_runoff_history_days to 2 years

ambrad committed:
Homme: Update Perlmutter machine file.

bartgol committed:
EAMxx: allow to reset fields/filename in AtmosphereInput

bartgol committed:
EAMxx: first attempt at DataInterpolation class

* The class handles time, horizontal, and vertical interpolation
* The vertical/horizontal are optional, but the time dimension is REQUIRED

grnydawn committed:
Merge branch 'ykim/frontier/scream-cime-merge' (PR #6771)

* Changed from Core/24.07 to Core/24.00
* Changed from cmake/3.27.9 to cmake/3.21.3
* Restored '--allow-shlib-undefined --allow-multiple-definition' in CMAKE_EXE_LINKER_FLAGS
* Removed 'craype-accel-amd-gfx90a' module load for amdclanggpu_frontier
* Added two mosart source files into NOOPT_FILES to workaround to prevent optcg crayclanggpu build error

[BFB] No baseline for Frontier yet.

jonbob committed:
Add call to zero out x2g_gx if the accum counter is 1

jonbob committed:
Add l2gacc_lx_cnt_avg to count number of times used in averaging

jonbob committed:
Use l2gacc_lx_cnt_avg for calculating some glc budget terms

chloewhicker committed:
mods to QICE long name

jgfouca committed:
minor fixes

rljacob committed:
Add ATM2ICE_FMAPNAME_NONLINEAR to moab coupler

Add ATM2ICE_FMAPNAME_NONLINEAR to moab coupler. Needed to create
ne30 cases.

oksanaguba committed:
Merge 'ABHI/abagusetty/homme/kokkos' (PR #6726)

Remove a removed kokkos command line flag not applicable anymore
Addresses the print statements from each rank

[bfb]

jinboxie committed:
Minor edits on orodrag documentation.

	modified:   tech-guide/orodrag.md
	modified:   user-guide/namelist_parameters.md
[BFB]

mahf708 committed:
remove obsolete inputdata testing script

mahf708 committed:
EAMxx: rework horizontal contraction utility impl

jonbob committed:
Add logic to make sure glc_nec is at least one

jonbob committed:
Fix trailing white space

jinboxie committed:
Minor text fix.
	modified:   orodrag.md
[BFB]

rljacob committed:
Merge branch 'maint-3.0' (PR #6777)

Bring changes from maint-3.0 to master. These include

Modifications to pe-layouts on anvil and improv for v3 cases.
Several updates to the moab coupler including:

Simplify the land grid instantiation for moab driver
fix errors related to aream initialization for rof model
Add a moab coupler developer test suite and fix bugs found by that suite

[BFB]

bartgol committed:
EAMxx: set valgrind supp file in ghci-snl mach file

bartgol committed:
EAMxx: suppress cmake warning

jonbob committed:
Better way to handle call to prep_glc_zero_fields

matthewhoffman committed:
Modify TFocean vertical interpolation for depths outside of data range

The ISMIP6 code for interpolating TF vertically does not give correct
values for depths below the deepest data z-level.  This commit handles
that case by using the deepest TF value corrected by the freezing
temperature depth-dependence.  It also handles the less critical
situation of depths above the shallowest data level by using the
shallowest value unmodified.

bartgol committed:
Merge branch 'bartgol/eamxx/ghci-set-valg-supp-file' (PR #6781)

This will allow us to use a pre-existing supp file, rather than relying on EKAT to build it on the fly. The reason is twofold: first, the simple program that EKAT uses to generate the file only picks up one of the many valg errors we get; second, and more important, the current EKAT infrastructure generates an empty file...

We will modify the supp file for ghci-snl when new suppressions are needed. For now, I have it to contain openmpi-valgrind.supp (shipped by our mpi installation) plus a few suppressions I generated running some eamxx tests...

bartgol committed:
EAMxx: add tests for data interpolation

bartgol committed:
Merge branch 'bartgol/eamxx/vert-remapper-refactor' (PR #6770)

In view of unifying the "dataset interpolation" implementations across eamxx, this PR makes the VerticalRemapper a bit more general. Namely:

- can be constructed from two existing grids (as opposed as assuming tgt grid is from map file);
- can handle "P0" and "Masked" extrapolation (the former does constant extrapolation, the latter sets extrapolated value to a constant mask value)
- can do different extrapolations at top and bottom of the column (but same extrapolation type across all columns)
- can handle 1d as well as 2d pressure profile, for both source and target pressure

mahf708 committed:
EAMxx: make horizontal contraction utility stricter

jonbob committed:
Merge branch 'xylar/ocn-ice/add-scaled-dismf-dib' (PR #6696)

Add scaling of DIB and DISMF by running mean of removed Antarctic ice runoff

This merge adds a capability by which data iceberg (DIB) freshwater
fluxes and data ice-shelf melt fluxes (DISMF) can be scaled by the
running mean of the the amount of ice (solid) runoff coming from
Antarctica (which is removed from the system by MPAS-Ocean).

This merge includes:
* A capability in MPAS-Ocean to compute daily averages of the area
  integrated removed ice runoff
* A capability in MPAS-Ocean to compute the running mean of these daily
  means over a given period of time (currently 2 years = 730 days)
* A capability to pass this running mean through the coupler (via the
  field rmean_rmv_ice_runoff in the infodata datastructure) from
  MPAS-Ocean to MPAS-Seaice
* A requirement that the the total annual mean DIB and DISMF fluxes be
  computed and stored in the DIB and DISMF input files. These files are
  currently only available for the IcoswISC30E3r5 mesh.
* Capabilities to compute a scale factor between the annual mean total
  data freshwater flux and the running mean of the removed ice runoff in
  both MPAS-Ocean and MPAS-Seaice
* A capability to scale landIceFreshwaterFlux and landIceHeatFlux in
  MPAS-Ocean by the scale factor
* A capability to scale bergFreshwaterFlux and bergLatentHeatFlux in
  MPAS-Seaice by the scale factor

[NML]
[BFB] stealth feature

mahf708 committed:
remove unused header id info

darincomeau committed:
Changing file name again

bartgol committed:
Merge branch 'mahf708/eamxx/field-colred' (PR #6776)

Adds a horizontal contraction utility equivalent to einsum('i,i...k->...k', weight, field)

jonbob committed:
First shot to make chicoma-gpu look like chicoma-cpu

bartgol committed:
Merge branch 'mahf708/eamxx/gh-ci-cleanup' (PR #6775)

Upgrades the underlying container for testing, remove redundancies and improve things due to the EAMxx merge-in.

mahf708 committed:
Workflows: remove mkatmsrffile test

jgfouca committed:
FortranData should no longer be using lviews

ambrad committed:
Homme: Update perlmutter machine file with PR 6423 fix.

bartgol committed:
EAMxx: removed pointless CMake lines

matthewhoffman committed:
Fix indexing errors

jgfouca committed:
Fixes for minor test mistakes

matthewhoffman committed:
Fix new indexing error introduced trying to fix indexing errors

mahf708 committed:
EAMxx: add horizontal average diagnostic field

matthewhoffman committed:
Improve error checking in ocean extrap

darincomeau committed:
Increasing range of AIS ice removal to 45S

ambrad committed:
Hommexx: Rework skipping timers in first step.

Make this a cmake define triggered by 'SYCL_BUILD'. Move it outside of the
threaded region for consistency with t_dis/enablef usage. Switch based on
nstep==0. Use a local boolean to avoid repeated calls to t_enablef. (The define
prevents any of this code from mattering in all cases except the intended
application.)

bartgol committed:
Delete .mergify.yml

While we tried to use this in the SCREAM repo to automize merging and to guard against accidental merges, we are not using it in the E3SM repo, and it is confusing at best.

rljacob committed:
Merge brancdh 'mahf708/ig/remove-unused-ci-job' (PR #6783)

Removes an unused test so that it is not accidentally triggered, and we also don't want to maintain it.

[BFB]

rljacob committed:
Merge branch 'sarats/share/pacer' (PR #6772)

Add Pacer, a timing interface for C++ components in E3SM such as Omega. Presently, it provides a thin wrapper around
GPTL with comprehensive error checking. In addition to providing basic timer functionality, it keeps track of open timers
and issues warnings about dangling timers at the end.

To begin with, the build configuration will be added to standalone Omega builds. We will add necessary CMake logic for
coupled E3SM builds in the future once Omega is integrated into E3SM main repo.

It just adds the sources under share/pacer and would not impact E3SM's builds at this point.
[BFB]

bartgol committed:
Merge branch 'bartgol/delete-mergify-config' (PR #6793)

While we tried to use this in the SCREAM repo to automize merging and to guard against accidental merges, we are not using it in the E3SM repo, and it is confusing at best.

mjschmidt271 committed:
rebase and fix conflicting variable usage regarding (m_)fields_inited

jonbob committed:
Merge branch 'darincomeau/ocn/add-sorrm-output' (PR #6757)

Turn on additional output for SORRMv3 mesh

Turns on the ocean conservation analysis member and ocean eddy analysis
member for the SOwISC12to30E3r3 mesh. This allows plotting of mass flux
anomaly due to land ice fluxes and EKE in MPAS-Analysis.

[BFB]

jonbob committed:
Merge branch 'sbrus89/ocn/rk4-time-fix' (PR #6695)

Evaluate time-dependent tendencies at correct time for each RK4 stage

This PR corrects a longstanding issue with RK4 and tendency terms which
have an explicit time dependence. Big thanks to @gcapodag and
@jeremy-lilly for providing the fix in their LTS development branch.

Here, I have migrated these changes from their MPAS-Dev branch to E3SM
and added the modifications to correct the convergence issues for the
manufactured solution test case

[BFB] - mpaso standalone

jgfouca committed:
Merge branch 'jgfouca/bfb_unit_no_f90_shoc' into master (PR #6779)

Following the pattern of a recent P3 PR, this PR changes the bfb unit
tests to be based on baseline files instead of fortran calls. Unlike
in that PR, we are actually able to completely remove all dependence
on fortran code here because SHOC doesn't need any f90 init routines.

Change list:
1) Main change: For all the BFB SHOC unit tests, use baseline files instead of calling fortran!
2) Remove all of the CXX->f90 bridge code
3) Cleanup of shoc_init situation. SHOC has no global data, so you only need to call it if you need to compute npbl
4) Reorg by moving some less-interesting testing stuff into shoc/tests/infra. Now, the files presented in the public interface (just shoc directory) is a lot cleaner.
5) All shoc tests (bfb and run_and_cmp) accept the same set of arguments (-c for compare, -g for generate, -b for baseline dir, and -n for no baselines).
6) Some of the tests that just confirm that DIFFs are caught do need to do thread count spreads.
7) Remove the ability of shoc.F90 to call CXX impls..
8) gen-boiler can now generate _host functions for function that have array data.
9) No longer need to worry about transposing array data, we never cross language boundary anymore.

There were a lot of f90 functions that had a property test but were never actually used in the CXX. I removed all of these:
* shoc_impli_sfc_fluxes_tests.cpp
* shoc_impli_srf_stress_tests.cpp
* shoc_impli_srf_tke_tests.cpp
* shoc_energy_total_fixer_tests.cpp
* shoc_energy_dse_fixer_tests.cpp
* shoc_energy_threshold_fixer_tests.cpp
* shoc_fterm_input_third_moms_tests.cpp
* shoc_fterm_diag_third_moms_tests.cpp
* shoc_omega_diag_third_moms_tests.cpp
* shoc_xy_diag_third_moms_tests.cpp
* shoc_aa_diag_third_moms_tests.cpp
* shoc_w3_diag_third_moms_tests.cpp

[BFB]

Moved over from: E3SM-Project/scream#3121

P3 DIFFs are expected because I caught a few places that were transposing views that didn't need to do that anymore. SHOC fails are expected because the baselines haven't been generated for the containers yet.

Follow on work:

Port P3 init code from fortran; this is the last of the fortran being used by either p3 or shoc
Remove all references to "fortran" or "Fortran". These names are all obsolete.

darincomeau committed:
Changing threshold to 57S

amametjanov committed:
Rename PE-layouts: S-20n, M-40n, L-64n.

amametjanov committed:
Move SYCL link flags to OMEGA_SYCL_EXE_LINKER_FLAGS

whannah1 committed:
fix MMF build issues

amametjanov committed:
Merge branch 'azamat/chrysalis/v3RRM-PEs' (PR #6700)

v3.NARRM PE-layouts on Chrysalis

v3.NARRM WCYCL PE-layouts on Chrysalis:
- Tiny:    10 nodes -- ~1 sypd
- Small:   20 nodes -- ~2 sypd
- SMedium: 30 nodes -- ~3 sypd
- Medium:  40 nodes -- ~4 sypd
- XMedium: 50 nodes -- ~5 sypd
- Large:   64 nodes -- ~6 sypd

Also, on Anvil:
- Small:   64 nodes -- ~1.8 sypd
- Medium:  96 nodes -- ~2.5 sypd
- Large:  128 nodes -- ~3.2 sypd

[BFB]

amametjanov committed:
Merge branch 'azamat/oneapi-ifxgpu/update-link-flags' (PR #6715)

Update CMake for oneapi-ifxgpu runs

Move -Xsycl-target-backend to link flags.

[BFB]

cbegeman committed:
Bug fixes from Hyun's code review

Co-authored-by: Hyun (Hyun-Gyu) Kang <47987430+hyungyukang@users.noreply.github.com>

jgfouca committed:
Convert p3_init from f90 to CXX

This removes the last f90 from P3.

Change list:
* Move table read/write implementation to CXX in p3_init_impl.hpp
* p3_init now returns a P3LookupTables object with all the views
  inside containing the tables.
* Rename init_kokkos_tables to get_global_tables
* Rename init_kokkos_ice_lookup_tables to get_global_ice_lookup_tables
* Remove all f90 bridge code
* p3_tables_setup is now built by default and ran when baseline gen
  is on. This gives us the ability to generate these tables through
  GH actions which will be needed for this PR.
* Move p3_tables_setup to tests dir, cleans up top p3 directory
* Fix p3_tables_setup implementation; it got broken in recent PRs
  but we didn't notice because it didn't get built by default
* Remove P3GlobalForFortran test struct, just call p3_init!
* Remove P3InitAP3Data test struct

Debatable change:
I really wanted to have `P3LookupTables lookup_tables` be static in
p3_init so they would not have to be re-created on every p3_init call.
Unfortunately, Kokkos does not allow static views to be active on program
shutdown, since this occurs after Kokkos::finalize, so we would need add
a p3_finalize method and ensure it was called at the end of p3. This
seems burdensome and error prone, so I just decided to re-read all the
tables every time p3_init is called.

[non-BFB]

bartgol committed:
EAMxx: add a grids manager from pre-built grids

This GM is useful for unit tests, but more generally for situations
where only grids are avaialble, but interfaces require a grids manager.

bartgol committed:
EAMxx: factor diagnostics creation out of IO class

* Add free function to build a diagnostic obj from a diag field name
* Adapt existing diags so that their name is "static", rather than
  matching the diag field name
* Adapt AtmosphereOutput to use this utility. Enhances encapsulation,
  and avoids hiding the diag naming convention deep in the IO class

jonbob committed:
Add perrWith=quiet for So_tf2d to avoid issues when it is not active

jonbob committed:
Get and use ocn_c2_glctf from infodata in ocn and glc drivers

Commit:
change default value of lambda_high in SHOC to 0.08

jonbob committed:
Merge branch 'darincomeau/ocn/change-sorrm-ice-runoff' (PR #6759)

Change SORRMv3 ice runoff mapping

Changes the ice runoff mapping file (ROF2OCN_ICE_RMAPNAME) for the
SOwISC12to30E3r3 ocean mesh to version with increased smoothing.

New mapping file has been staged in public inputdata repo, world
readable.

[BFB] for all currently tested configurations

wlin7 committed:
Merge branch 'jinboxie/atm/e3smv3_oro_merge' (PR #6667)

Implementation of new orographic drag parameterization scheme to alleviate wind bias in E3SM

Add a new suite of orographic drag parameterization schemes into EAM.
It includes 4 components all combined in one module (i.e. subroutine gwdo2d).
The schemes include orographic gravity wave drag (oGWD, Xie et al.,2020),
flow-blocking drag (FBD, Xie et al.,2020), small-scale GWD (sGWD, Tsiringakis et al.,2017),
and turbulent-scale orographic form drag (TOFD, Beljaars et al., 2004).

The scheme is distributed by the interface named gwdo_gsd. The oGWD, FBD, and
sGWD are implemented in gw_drag.F90, while TOFD is implemented in clubb_intr.F90
to join the vertical diffusion process. Another change is in the startup_initialconds.F90
and gw_drag.F90 where the new topographic parameters (OA, OC, OL) are input into the model.

Although it is possible to turn each new scheme on/off individually, the only validated
config is to have them all on (and the original scheme off).

Technical guide for this scheme is available at
https://docs.e3sm.org/E3SM/EAM/tech-guide/orodrag/

[BFB]

whannah1 committed:
initial implementation of relaxed slab ocean mode

jgfouca committed:
Fix conditional jump on uninitialized mem in SHOC

The rino temp view only had the nlev-1 entry initialized. pblintd_height was
accessing other entries.

bartgol committed:
EAMxx: add QOL methods to TimeStamp

Simplifies some operations that looked clunky on the client end

bartgol committed:
EAMxx: fix valgrind tests

jgfouca committed:
Remove used var

whannah1 committed:
update PAM submodule

jgfouca committed:
Fix and add comment

jgfouca committed:
Restore old code if ndebug not on

Commit:
modify default lambda_high value in shoc tech doc file

jonbob committed:
Updates for the glc budget

tcclevenger committed:
Remove unnecessary if-block

if (iop_nudge_tq or iop_nudge_uv) {
  ...
  if (iop_nudge_tq or iop_nudge_uv) {
    ...
  }
}

tcclevenger committed:
Use field util horiz_contraction to compute means

bartgol committed:
Merge branch 'bartgol/eamxx/encapsulate-diagnostics-creation' (PR #6796)

Extract creation of diagnostic classes out of the AtmosphereOutput class and into its own routine.

jgfouca committed:
Fix bug in p3/shoc run_and_cmp

The for loop the processes options was stopping too soon. I
think maybe someone assumed that every option would have a value
after it, but that is no longer true. Also, the -n (no-baseline) option
conflicted with the nadv option for shoc, so I changed that to -l

jgfouca committed:
Merge branch 'jgfouca/fix_shoc_uninit_mem' into master (PR #6799)

Fix conditional jump on uninitialized mem in SHOC

The rino temp view only had the nlev-1 entry
initialized. pblintd_height was accessing other entries.

I looked the the F90 impl and it is also doing this strange
initialization of only nlev-1.

This PR will fix valgrind errors for shoc_run_and_cmp that have
appeared on our dashboard. This error seems to go back a long time, so
I don't understand why it only showed up now.

[BFB]

jgfouca committed:
Merge branch 'jgfouca/fix_run_and_cmp' into master (PR #6805)

Fix bug in p3/shoc run_and_cmp

The for loop that processes options was stopping too soon. I think
maybe someone assumed that every option would have a value after it,
but that is no longer true. Also, the -n (no-baseline) option
conflicted with the nadv option for shoc, so I changed that to -l.

This PR should fix the nightly fails for shoc_run_and_cmp.

[BFB]

* origin/jgfouca/fix_run_and_cmp:
  Fix bug in p3/shoc run_and_cmp

stephenprice committed:
Merge branch 'E3SM-Project:master' into stephenprice/glc/update-glc-budgets-new

wlin7 committed:
Merge branch 'whannah/atm/fix-MMF2-bld' (PR #6801)

update PAM submodule to fix build error

The PAM update removes include lines for the SCREAM *_f90.hpp files that have been removed.

[BFB]

jonbob committed:
Merge branch 'cwhicker/elm/smb_h0_fields' (PR #6682)

Introduce land ice surface mass balance fields in default runs without MECs/GLC

Introduced new diagnostic Surface Mass Balance fields, they are
calculated as if there is a coupled land ice component but are only
diagnostic - it does not modify the energy/water budgets in default
or coupled GLC simulations.

[BFB]

bartgol committed:
EAMxx: add a TimeInterval lightweight struct

jgfouca committed:
Init rino when short tests are on

The mem tests are no longer guaranteed to have NDEBUG on. We
use SCREAM_SHORT_TESTS as a proxy for knowing that mem checking
is on.

bartgol committed:
EAMXX: fix misc compiler warnings

stephenprice committed:
Minor cleanup

Remove placeholder / commmented out code in glc_comp_mct

Correct revision history comment in glc section of seq_diag_mct

bartgol committed:
EAMxx: changes in VerticalRemapper

* Allow to set only one src or tgt pressure profile. This is because
  we may not have any mid or any int field. No need to force us to
  create a valid field just for passing checks.
* Allow to query the remapper for the src/tgt mid/int pressure profile

tcclevenger committed:
EAMxx: Unrelated: remove left over debug print from scorpio_output.cpp

jgfouca committed:
Merge branch 'jgfouca/fix_shoc_valg' into master (PR #6813)

The mem tests are no longer guaranteed to have NDEBUG on. We use
SCREAM_SHORT_TESTS as a proxy for knowing that mem checking is on.

[BFB]

* jgfouca/fix_shoc_valg:
  Init rino when short tests are on

glemieux committed:
increase character length for fates harvest mode

bartgol committed:
HOMME: fix some valgrind errors in namelist parsing module

bartgol committed:
EAMxx: use pack size 1 in valg build, to avoid false-positives from  padding

Similar to what we do for FPE builds

bartgol committed:
EAMxx: move DefaultMetadata dictionary to share/util

bartgol committed:
EAMxx: add possibility to dump the content of the FieldManager to a yaml file

jonbob committed:
Merge branch 'jonbob/machines/update-chicoma-gpu' (PR #6791)

Update chicoma-gpu machine files

Updates LANL IC platform chicoma-gpu machine files, after recent upgrade

Fixes #6778
[BFB]

tcclevenger committed:
Use packs for *_mean views

Fixes a bug where real views were being used as if they were packed.

mahf708 committed:
EAMxx: clarify error in field_at_ diagnostics

mahf708 committed:
EAMxx: simplify the testing of horiz_contraction

jonbob committed:
Clean-up ocn_c2_glc variables so they are correctly set by mpaso

jgfouca committed:
Github feedback

Simplify io action using if constexpr.
Remove fortran module stuff from cmake
Remove fortran from comment

bartgol committed:
Update ekat submodule

Pull in a better way of dumping ParameterList to yaml files

bartgol committed:
EAMxx: fix several compiler warnings

bartgol committed:
Merge branch 'E3SM-Project/bartgol/eamxx/valg-fixes' (PR #6841)

Fixes valgrind tests failures in eamxx standalone tests.

tcclevenger committed:
Merge branch 'bogensch/eamxx/shoc_lambda_high_default' (PR #6797)

This "new" value of 0.08 for lambda_high for SHOC has been used in all
recent SCREAM production runs (CESS experiments; decadal) but for some
reason was never made default in master. Correcting that here.
Modified from 0.04 to 0.08 because this improves the representation of
marine stratocumulus.
    
[non-BFB]

bartgol committed:
Merge branch 'mahf708/eamxx/horiz-avg-diag' (PR #6788)

Adds an online diagnostic field for area-weighted horizontal average, following #6776.

whannah1 committed:
add namelist vars for disabling P3 subgrid cldfrac

"disabling" might be misleading because these flags "maximize" cloud fraction by setting them to 1 everywhere

whannah1 committed:
make P3 cld frac flags more verbose

whannah1 committed:
add type and doc for namelist flags

iulian787 committed:
number of ghost layers needs to be set for bilinear maps

need a new version of moab master, anyway

bartgol committed:
Workflows: test MMF in eamxx v1 workflow

tcclevenger committed:
Remove unnecessary fence

mjschmidt271 committed:
fast-forward mam4xx submodule to fix non-determinism in amicphys test

bartgol committed:
Merge branch 'bartgol/eamxx/dump-fields-dictionary' (PR #6842)

The option (off by default) allows to save a yaml file containing a dictionary of the fields on the "Physics" grid, with some of their details.

whannah1 committed:
update RSO config and namelist variables

whannah1 committed:
refactor p3 cloud fraction flags

jonbob committed:
Move runningMeanRemovedIceRunoff from restart_contents to streams

whannah1 committed:
remove old method

bartgol committed:
EAMxx: add testing for horiz remap to data interpolation

jonbob committed:
Merge branch 'stephenprice/glc/update-glc-budgets-new' (PR #6634)

Update glc coupler budgets for active Greenland ice sheet

This PR adds functionality to coupler budget code so that x2g_ and l2x_
fields associated with Greenland ice sheet surface mass balance are
accounted for in the coupler budget tables. Also added, but currently
passing fields of zeroes, is support for g2x_ fields associated with
iceberg fluxes when running with a dynamic Greenland ice sheet. Includes
a correction to the renormalization scheme in prep_glc_mod.

[non-BFB] for configurations with active lnd and glc

whannah1 committed:
add refs for data model documentation

whannah1 committed:
add data model documentation

whannah1 committed:
linter fixes

whannah1 committed:
delete bibtex abstracts

jonbob committed:
Merge branch 'jonbob/seaice/fix-restart-contents' (PR #6857)

Fix issue with mpas-seaice restart_contents

A new field runningMeanRemovedIceRunoff was added in PR #6696 that
causes errors in some configurations on pm-cpu and frontier with the gnu
compiler. This moves the new field from a shared "meta-stream" to the
streams themselves because attached package wasn't getting inherited
correctly.

Fixes #6855
[BFB]

bartgol committed:
Merge branch 'bartgol/workflows/add-mmf-test-to-eamxx-v1' (PR #6852)

Allows to verify that MMF tests (relying on scream physics) are not broken by EAMxx PRs.

cbegeman committed:
Move manufactured soln parameter setting to init routine

mahf708 committed:
EAMxx: move cld_frac overrides to P3Runtime

bartgol committed:
EAMxx: reorg query-cf-database utility

* Moved to its own folder
* Do not add scripts subdir when building EAMxx
* Move data/*yaml in query-cf-database folder

bartgol committed:
EAM: removed stale code in buildnml

The folder eamxx/data no longer exists, and did not contain any input
data for EAM in the first place

ambrad committed:
EAM(xx): Add CIME EAM and EAMxx tests for SL enhanced trajectory method.

Also add parameters to namelist definition file.

bartgol committed:
EAMxx: fix issues in VerticalRemapper

We were still assuming tgt layout could only have LEV
and not ILEV.

bartgol committed:
EAMxx: allow to specify extrap type for vremap in DataInterpolation

mjschmidt271 committed:
ff mam4xx to agree with interface

fix work array pointers in mam4::wetdep::aero_model_wetdep

update input.yaml for standalone microphys

minor changes to get things running

tcclevenger committed:
Merge branch 'tcclevenger/eamxx/iop_as_atm' (PR #6787)

bartgol committed:
EAMxx: fix md linting of README

mahf708 committed:
docs: only run docs when docs are touched

bartgol committed:
EAMxx: some small fixes to horiz/refine remappers

bartgol committed:
EAMxx: fix p1d read in DataInterpolation

bartgol committed:
EAMxx: finally get DataInterpolation and its tests to work

jinboxie committed:
Fixed the namelist bug for changing tuning par in oro_drag.

	modified:   control/runtime_opts.F90
	modified:   physics/cam/clubb_intr.F90
	modified:   physics/cam/gw_drag.F90
	modified:   physics/cam/od_common.F90
	modified:   physics/cam/phys_control.F90

[BFB]

jonbob committed:
Merge branch 'darincomeau/ocn/change-ais-ice-removal' (PR #6794)

Change range of solid ice runoff removal from 60S to 57S

Extends the northern range of solid ice runoff from Antarctica from 60S
to 57S when the ocean's namelist option
config_remove_ais_ice_runoff = .true.. The new solid ice runoff mapping
with increased smoothing for the SOwISC12to30E3r3 mesh in PR #6759 has a
small amount of runoff north of 60S, which needs to be removed when
Antarctic solid ice runoff is intended to be removed.

[BFB] for all tested configuration

dependabot[bot] committed:
Bump DavidAnson/markdownlint-cli2-action from 18 to 19

Bumps [DavidAnson/markdownlint-cli2-action](https://github.com/davidanson/markdownlint-cli2-action) from 18 to 19.
- [Release notes](https://github.com/davidanson/markdownlint-cli2-action/releases)
- [Commits](https://github.com/davidanson/markdownlint-cli2-action/compare/v18...v19)

---
updated-dependencies:
- dependency-name: DavidAnson/markdownlint-cli2-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

rljacob committed:
Merge branch 'mahf708/docs/do-not-build-all-the-time' (PR #6865)

Only run docs workflows when docs are touched. This reduces the comments by the GitHub-actions [bot] on PRs.

[BFB]

whannah1 committed:
reorganize data ocean docs

whannah1 committed:
linter fixes

matthewhoffman committed:
Update zOcean error check

tcclevenger committed:
Merge branch 'whannah/emaxx/add-p3-cld-frac-flags' (PR #6849)

These flags change the input P3 cloud fraction by setting them to 1 everywhere. Currently 
they default to False, but we will likely enable them as part of the effort to address 
the popcorn convection problem.

[BFB]

matthewhoffman committed:
Update error message

Co-authored-by: Xylar Asay-Davis <xylarstorm@gmail.com>

jgfouca committed:
Change new table names to not conflict with older tables

matthewhoffman committed:
Merge branch 'matthewhoffman/mali/tf-vert-interp-fix' into MALI-Dev/develop

The ISMIP6 code for interpolating TF vertically does not give correct
values for depths below the deepest data z-level. This commit handles
that case by using the deepest TF value corrected by the freezing
temperature depth-dependence. It also handles the less critical
situation of depths above the shallowest data level by using the
shallowest value unmodified.

* origin/matthewhoffman/mali/tf-vert-interp-fix:
  Update error message
  Update zOcean error check
  Improve error checking in ocean extrap
  Fix new indexing error introduced trying to fix indexing errors
  Fix indexing errors
  Modify TFocean vertical interpolation for depths outside of data range

jonbob committed:
Merge branch 'matthewhoffman/mali/gis_ocn_to_glc_coupling' (PR #6632)

OCN to GLC thermal forcing coupling

This pull request introduces coupling from OCN to GLC to pass thermal
forcing at a prescribed ocean depth (300 m) from MPAS-Ocean to MALI,
where MALI uses it to calculate grounded marine melting through an
existing 'facemelting' parameterization. Facemelting as a function of
ocean thermal state is a critical OCN/GLC coupling for Greenland, but it
is relevant to Antarctica as well. This OCN-GLC thermal forcing coupling
is activated whenever OCN is present and GLC is active.

This PR has pieces in MPAS-Ocean, the coupler, and MALI:
* In MPAS-Ocean, a new avgThermalForcingAtCritDepth field is added and
  calculated. The depth at which to calculate the thermal forcing is
  namelist configurable. The default is 300 m.
* In MALI, the field passed from the coupler is collected in
  ismip6_2dThermalForcing and used in the existing facemelting
  parameterization. Namelist and streams are updated to use facemelting
  in all simulations and output relevant variables. For compsets where
  TF is not being coupled, it will have values of 0 and result in no
  facemelting being applied; it is safe to have this option generally
  turned on.
* In the coupler, a new remapping object is created that handles the TF
  mapping independently of any other fields passed from OCN to GLC. New
  mapping files are added. Any coupling variables related to the
  existing ice-shelf coupling now include the 'shelf' suffix and all
  coupling variables related to this new thermal-forcing coupling have
  the 'tf' suffix.
* A new compset is created that is a standard G-case but with active
  MALI. New mapping files are added between the relevant grids for the
  new mapper.
* The new OCN2GLC_TF_SMAPNAME mapping should be a nearest source to
  destination mapping with the MPAS-Ocean source mesh masked to only
  include grid cells that are deeper than the critical depth to avoid
  extrapolating undefined values. A tool to generate that map file is
  added to MPAS-Dev/MPAS-Tools
* Two new model_grids are added that are compatible with this feature
  and include the special mapping file required:
  TL319_IcoswISC30E3r5_gis1to10kmR2 and TL319_oQU240wLI_gis20
* The new coupling is controlled by a new namelist option in MPAS-Ocean:
    config_glc_thermal_forcing_coupling_mode
  It defaults to 'off' and there are no compsets that explicitly set it
  for now, making this a stealth feature.

[NML] for new mappings, new ocean variable
[non-BFB] for configurations with active MALI

jgfouca committed:
Update PAM submodule

bartgol committed:
Merge branch 'bartgol/eamxx/do-not-build-scripts-dir-with-cmake' (PR #6861)

Doing some cleaning: these files were built along with eamxx in certain conditions, despite being never used.

jgfouca committed:
GPU fixes: need to use host views for init stuff

ambrad committed:
Hommexx: Add a timer for vertical_remap of dynamics variables.

bartgol committed:
EAMxx: create rrtmgp allsky baseline target with CreateUnitTest utility

The cmake utility links (indirectly) against scream_io, which avoids
link errors in case of shared libraries. The errors are due to the
fact that scream_share does use scream_io stuff, but the rrtmgp
executable was not linking against it. Since the exec was indeed
not using ANY of the scream_io symbols, static linking is fine, but
dynamic linking is not.

ambrad committed:
Homme/SL: Adjust auto-setting of halo.

bartgol committed:
EAMxx: fix a couple of bugs related to time stamp

* The curr_month_beg method was using the wront time of day
* The next_write_ts method was not considering a corner case for
month/year frequence. Resetting to what was previously in master.

rljacob committed:
Add ocn_c2_glctf to seq_infodata as in mct coupler

add ocn_c2_glctf to seq_infodata_PutData_explicit and getData as
was done for mct coupler.  Needed to build with current MPAS

rljacob committed:
Update config_component_e3sm.xml from v3.0

Update config_component_e3sm.xml from v3.0 driver-mct

rljacob committed:
Add tri grid case to moab suite

Add fully coupled tri grid case to moab suite in addition to bi-grid

whannah1 committed:
adjust type specification for RSO mode

oksanaguba committed:
trying to bring sunspot entry in match with auroras

whannah1 committed:
further r8 updates and set fixed values as parameters

jgfouca committed:
Merge branch 'jgfouca/p3_cxx_init' into master (PR #6845)

This removes the last f90 from P3.

Change list:
1) Move table read/write implementation to CXX in p3_init_impl.hpp
2) p3_init now returns a P3LookupTables object with all the views inside containing the tables.
3) Rename init_kokkos_tables to get_global_tables
4) Rename init_kokkos_ice_lookup_tables to get_global_ice_lookup_tables
5) Remove all f90 bridge code
6) Move p3_tables_setup to tests dir, cleans up top p3 directory
7) Fix p3_tables_setup implementation; it got broken in recent PRs but we didn't notice because it didn't get built by default
8) Remove P3GlobalForFortran test struct, just call p3_init!
9) Remove P3InitAP3Data test struct

Debatable change:

I really wanted to have P3LookupTables lookup_tables
be static in p3_init so they would not have to be re-created on every
p3_init call. Unfortunately, Kokkos does not allow static views to be
active on program shutdown, since this occurs after Kokkos::finalize,
so we would need add a p3_finalize method and ensure it was called at
the end of p3. This seems burdensome and error prone, so I just
decided to re-read all the tables every time p3_init is called.

There are some round-off level diffs in P3, so:

[non-BFB]

* jgfouca/p3_cxx_init:
  GPU fixes: need to use host views for init stuff
  Update PAM submodule
  Change new table names to not conflict with older tables
  Github feedback
  Convert p3_init from f90 to CXX

whannah1 committed:
fix data comp docs error

bartgol committed:
Merge branch 'bartgol/eamxx/fix-rrtmgp-test-with-shared-libs' (PR #6875)

Fix issue in eamxx build system when we generate baselines using shared libs.

rljacob committed:
Merge branch 'iulian787/ghost_layers_bilinear' (PR #6851)

number of ghost layers needs to be setup correctly for the bilinear maps;
different PE layouts should result in the same map.

[BFB]

jonbob committed:
Merge branch 'cbegeman/ocn/enhance-manufactured-solution' (PR #6862)

Add del2 and del4 terms to manufactured solution test

This feature is used only for MPAS-Ocean standalone testing of the
horizontal mixing operators del2 and del4. It is an extension of the
existing manufactured solution test case. When either or both of del2
and del4 momentum terms are included in the dynamics, we add the
corresponding source terms to the manufactured solution at runtime.

[BFB] not used by E3SM

bartgol committed:
Merge branch 'eagles-project/mjs271/mam4xx/ff-submodule' (PR #6867)

This PR fixes the MAM4xx interface for wet scavenging--namely, addressing failing tests and an associated non-determinism issue.

The issues existed primarily on the MAM4xx side of things, so this PR is mostly a fast-forward of the MAM4xx submodule. However, minor edits to the EAMxx interface code were required in order to properly call mam4::wetdep::aero_model_wetdep().

rljacob committed:
Merge  branch 'rljacob/mbcpl/misc-fixes' (PR #6877)

Fix a few issues including:

Add ocn_c2_glctf to seq_infodata as in mct coupler which is needed after merging 31d8b7b
Add ATM2ICE_FMAPNAME_NONLINEAR to moab coupler. Needed to create
ne30 cases
Add a tri-grid case to moab dev test suite.
Update config_component_e3sm.xml with latest driver-mct version.

[BFB]

ambrad committed:
Homme/SL: Remove CAAS-point impl for now.

I'll add this in a different PR. However, keep the generalized indexing changes
and extra argument in the Homme interface because they are useful.

ambrad committed:
Hommexx/SL: Break main impl file into three.

oksanaguba committed:
make sunspot macros file ident to aurora

bartgol committed:
EAMxx: temporarily disable failing mam tests

oksanaguba committed:
make sunspot file ident to auroras

rljacob committed:
Merge branch 'dependabot/github_actions/DavidAnson/markdownlint-cli2-action-19' (PR #6872)

Bump DavidAnson/markdownlint-cli2-action from 18 to 19

ambrad committed:
Homme/SL: Rearrange some code; add more comments.

bartgol committed:
Merge branch 'eagles-project:mjs271/eamxx/DAG-refactor' (PR #6860)

Finishes PR #3101 originally submitted to the scream repo the atmospheric DAG generator (atmosphere_process_dag.xpp) and atmosphere driver to get things working again

mjschmidt271 committed:
update mam4xx for wetdep KOKKOS_INLINE_FUNCTION fix

bartgol committed:
Workflows: run eamxx workflows when mam4xx/haero are updated

bartgol committed:
Merge branch 'bartgol/workflows/run-eamxx-workflows-when-mam-changes' (PR #6887)

We need to test eamxx when mam4xx and/or haero submodules change.

jonbob committed:
Merge branch 'vanroekel/ocean/kpp-interface-fixes' (PR #6708)

Fixes bugs in CMix/KPP interface

Two parts of the KPP interface had bugs
* The computation of surface buoyancy forcing and friction velocity were
  incorrect. The surfaceBuoyancyForcing subtracted the wrong values from
  the heat fluxes from thickness fluxes. The surfaceFrictionVelocity had
  incorrect interpolation to centers, the interpolation itself was
  unneeded. MPAS was interpolating stress to edges (for momentum
  forcing) but interpolating back (instead of using coupler values) for
  mixing friction velocity.
* The KPP interface for matchBoth is fixed and the indexing is fixed for
  matching interior to boundary layer diffusivities.

[CC]

mahf708 committed:
EAMxx: add vert_contraction field utility

bartgol committed:
Merge branch 'eagles-project:mjs271/fix/m4x-kokkos-decorator' (PR #6886)

Update mam4xx submodule to address an issue with gpu/cpu function decorators and to correct the initialization of isprx in wetdep.

[BFB]

bishtgautam committed:
Merge branch 'lnd/fates-api37' (PR #6762)

Updates fates to API37 per a fates-side fire refactor which renames and moves `nfsc`.
This updates the fates tag with a number of fates-side fixes and feature updates so
fates tests will be answer changing.

Fixes #6806
[non-B4B] for fates tests

wlin7 committed:
Merge branch 'jinboxie/atm/oro_bugfix' (PR #6869)

Bug fix for changing orodrag tuning parameters via namelist

This is a bug fix for the latest orodrag parameterization
merged into master. The setup of the parameter tuning for
the schemes is intended to read in orodrag parameters such
as od_ls_ncleff, od_bl_ncd, od_ss_sncleff into the model for
tuning via phys_control.F90. However, the orodrag parameters
are set in a separate place spot in atm_in (e.g. oro_drag_nl),
and thus a separate oro_drag_readnl is added in od_common.F90
to read these parameters in atm_in. It is tested and works well
both use the new od schemes or the default schemes.

[BFB]

bartgol committed:
EAMxx: disable mam4xx wetscav tests

bartgol committed:
EAMxx: fix compiler warning

bartgol committed:
EAMxx: fix memory error in the atm proc dag generation

bartgol committed:
Merge branch 'bartgol/eamxx/disable-mam-test' (PR #6882)

Allows other PRs to test normally while the MAM team investigates further

mahf708 committed:
EAMxx: improve vert_contraction sanity checks

mahf708 committed:
EAMxx: add ILEV testing for vert_contraction

whannah1 committed:
bug fix for SCM tests

bartgol committed:
Merge branch 'bartgol/eamxx/fix-dag-valg-errors' (PR #6890)

The issue was simply that we were grabbing a ref to an element in a vector that was later resized, invalidating the ref.

mahf708 committed:
EAMxx: refactor vert_contraction to lessen code dupl

mjschmidt271 committed:
update mam4xx to include fix for wet scavenging

bartgol committed:
EAMxx: set testing max ranks to 2 on ghci-snl-cuda

mjschmidt271 committed:
re-enable wet scavenging test for mam

bartgol committed:
Merge branch 'bartgol/eamxx/data-interpolation' (PR #6812)

The class is in charge of reading in time-dependent datasets from input file(s), interpolate to current model time, and possibly do horizontal/vertical interpolation

mjschmidt271 committed:
ff mam4xx to main after merge

bartgol committed:
Merge branch 'mahf708/eamxx/vert-contraction' (PR #6889)

Adds a vertical contraction utility equivalent to einsum('k,...k->...', weight, field) or einsum('ik,i...k->i...', weight, field)

rljacob committed:
Merge branch 'whannah/docn/add-relaxed-slab-ocean' (PR #6798)

The relaxed slab ocean (RSO) is implemented as a simpler alternative to the traditional slab ocean
mode that requires externally generated heat flux data that represents the effects of ocean transport
on ocean surface temperature. In RSO mode we want to relax to a target SST field using only a
specified mixed layer depth to allow things like cold wakes following a tropical cyclone.

Also add documentation for this mode of the data ocean model.

[BFB]

bartgol committed:
EAMxx: fix compilation error on chrysalis

ambrad committed:
Merge branch 'ambrad/homme/isl2pr1' (PR #6874)

This PR brings in a new feature that (1) increases accuracy of semi-Lagrangian
tracer transport's trajectory calculations and (2) permits flexible trade-off
between the trajectory accuracy and speed. This PR has the following parts:

- F90 dycore support, with unit tests (principally in sl_advection.F90);
- C++ dycore support, with unit tests (principally in
  ComposeTransportImplEnhancedTrajectory.cpp);
- unit test driver updates: compose_ut.cpp;
- two new standalone-Homme tests, one each for the F90 and C++ dycores;
- new standalone-Homme transport test module for convergence testing: fully 3D,
  space-and-time-dependent surface pressure (dcmip2012_test1_conv_mod.F90);
- CIME-based ERS tests, one each, for EAM and EAMxx;
- cleanup of a timer issue orthogonal to this PR: see commit 'Hommexx: Rework
  skipping timers in first step.';
- updates to Homme machine files for Perlmutter;
- fix C++ dycore's handling of prescribed winds: had to move down in the call
  stack to match the F90 dycore.

e3sm_developer and e3sm_atm_integration pass on Chrysalis. EAMxx test suites
pass on Perlmutter and Frontier. There are no performance effects when the
stealth feature is off based on tests on Chrysalis (v3.LR 11-year control run),
Frontier, and Perlmutter.

[non-BFB] due to two new CIME tests, two new standalone-Homme tests, and two
modified standalone-Homme tests; otherwise BFB.

jonbob committed:
Make bld files consistent with Registry

bartgol committed:
Merge branch 'eagles-project:mjs271/mam4xx/resolve_wetscav' (PR #6894)

This brings in changes to mam4xx's wet_dep.hpp that correct the size of a work array
to fix out-of-bounds errors that have led to recent fails for the mam4_wetscav_standalone test.

bartgol committed:
EAMxx: fix bug in io_se_grid unit test

bartgol committed:
Merge branch 'bartgol/eamxx/fix-chrysalis-compiler-error' (PR #6896)

Fixes a compilation error that is most likely just a compiler parsing issue. Still, the change is innocuous.

bartgol committed:
Merge branch 'bartgol/eamxx/ghci-snl-cuda-max-rank-2' (PR #6895)

Set testing max ranks to 2 on ghci-snl-cuda

bartgol committed:
EAMxx: change extrapolation of SPA fields at the top of the atm

Do constant extrap rather than linear ramp down to 0 at p=0

bartgol committed:
EAMxx: fix misc compiler warnings

jgfouca committed:
Begin to de-scream things (use eamxx instead)

This PR changes the test suite names and unifies
the testmods under eamxx.

[BFB]

jgfouca committed:
eammx-prod only works with pg2 grids

bartgol committed:
Merge branch 'bartgol/eamxx/spa-change-tom-bc' (PR #6902)

Do constant extrapolation rather than linear ramp down to 0 (for p=0).

[non-BFB] for EAMxx

oksanaguba committed:
fix sunspot run error in internal_init mpi

jonbob committed:
Add missing file to landice.cmkae file list

jgfouca committed:
Fix cime-nml test that was using old testmod dir

jgfouca committed:
Merge branch 'jgfouca/begin_to_descream' into master (PR #6901)

This PR changes the test suite names and unifies the testmods under
eamxx.

We'll have to remember to update E3SM_test_scripts once this is
in. The dashboard will also need adjustments.

This PR also adds a couple eamxx v1 tests to e3sm_developer including a prod test.

[BFB]

* jgfouca/begin_to_descream:
  Fix cime-nml test that was using old testmod dir
  eammx-prod only works with pg2 grids
  Begin to de-scream things (use eamxx instead)

jonbob committed:
Merge branch 'matthewhoffman/mali/ocean-extrap-and-rk-fixes' (PR #6873)

Add MALI ocean extrapolation and fix to RK scheme

This PR updates the version of MALI from the MALI-Dev fork and primarily
brings in two important features needed for Greenland coupling work:
* ability to extrapolate ocean thermal forcing from a masked region of
  valid ocean data to the current MALI ice extent
* a critical bug fix for the RK timestepping scheme required for it to
  work correctly
* changes to handling of masks in the RK and FCT schemes to keep them
  consistent
Also included are some changes to the subglacial hydrology model, which
is not currently used in E3SM simulations.

[NML] for configurations with MALI
[non-BFB] for configurations with MALI

oksanaguba committed:
Merge branch 'E3SM/master' into oksanaguba/spot

 Conflicts:
	cime_config/machines/cmake_macros/oneapi-ifxgpu_aurora.cmake
	components/eam/src/physics/rrtmgp/external
	components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.hpp
	components/homme/cmake/HommeMacros.cmake
	components/homme/src/preqx_kokkos/cxx/cxx_f90_interface_preqx.cpp
	components/homme/src/share/cxx/ExecSpaceDefs.cpp
	components/homme/src/share/cxx/SimulationParams.hpp
	components/homme/src/theta-l_kokkos/cxx/cxx_f90_interface_theta.cpp
	components/homme/src/theta-l_kokkos/prim_driver_mod.F90
	externals/ekat
